src/Files.App/Dialogs/AddItemDialog.xaml.cs

```diff
@@ -34,8 +37,8 @@ private void ListView_ItemClick(object sender, ItemClickEventArgs e)
 
 		private async void AddItemDialog_Loaded(object sender, RoutedEventArgs e)
 		{
-			var itemTypes = await ShellNewEntryExtensions.GetNewContextMenuEntries();
-			await ViewModel.AddItemsToList(itemTypes); // TODO(i): This is a very cheap way of doing it, consider adding a service to retrieve the itemTypes list.
+			var itemTypes = await addItemTypesService.GetNewEntriesAsync();
+			await ViewModel.AddItemsToList(itemTypes);
```

Comment:
Should we do this for the right click menu as well?

Reply:
I think we should. I'll try to look into this

Reply:
At the moment, right click menu entries depends on right clicked items. This may be out of my reach, but nonetheless I'll try  to refactor the code to find a solution

Reply:
I meant for the add item sub menu, not everything else 👍

Reply:
Oh ok, I should have done that with `NewItem caching` commit